extensions/amp-analytics/0.1/events.js

```diff
@@ -380,6 +399,78 @@ export class CustomEventTracker extends EventTracker {
   }
 }
 
+class AmpStoryEventTracker extends EventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+
+    /** @private {?Observable<!Event>} */
+    this.sessionObservable_ = new Observable();
+
+    /** @private {?function(!Event)} */
+    this.boundOnSession_ = this.sessionObservable_.fire.bind(
+      this.sessionObservable_
+    );
+
+    Object.keys(StoryAnalyticsEvent).forEach(key => {
+      this.root
+        .getRoot()
+        .addEventListener(StoryAnalyticsEvent[key], this.boundOnSession_);
+    });
+  }
+
+  /** @override */
+  dispose() {
+    const root = this.root.getRoot();
+    Object.keys(StoryAnalyticsEvent).forEach(key => {
+      root.removeEventListener(StoryAnalyticsEvent[key], this.boundOnSession_);
+    });
+    this.boundOnSession_ = null;
+    this.sessionObservable_ = null;
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    const storySpec = config['storySpec'] || {};
+    userAssert(!config['selector'], 'Selector not supported in AmpStorySpec');
+    const selector = ':root';
+    const selectionMethod = config['selectionMethod'] || null;
+    const targetReady = this.root.getElement(
+      context,
+      selector,
+      selectionMethod
+    );
+    const repeat = storySpec['repeat'];
+    const on = config['on'];
+
+    return this.sessionObservable_.add(event => {
+      const {type} = event;
+
+      if (type !== on) {
+        return;
+      }
+      const details = /** @type {?JsonObject|undefined} */ (getData(event));
+      const detailsForPage = details['detailsForPage'];
+
+      if (repeat === false && detailsForPage['repeated']) {
+        return;
+      }
+
+      const el = dev().assertElement(
+        event.target,
+        'No target specified by story session event.'
+      );
+      targetReady.then(target => {
+        if (target.contains(el)) {
```

Comment:
And if the target is always guaranteed to be root, do we still need to check for `target.contains()`?

Reply:
Nope! Removed it.